operator/v1/types_ingress.go

```diff
@@ -716,8 +720,58 @@ type AWSNetworkLoadBalancerParameters struct {
 	// +optional
 	// +openshift:enable:FeatureGate=IngressControllerLBSubnetsAWS
 	Subnets *AWSSubnets `json:"subnets,omitempty"`
+
+	// eipAllocations assign Elastic IP addresses to the Network Load Balancer.
+	// eipAllocations cannot be used with "Internal" scope.
+	// An EIP can be allocated to only a single IngressController.
+	// The number of Allocation IDs must match the number of subnets that are used for the load balancer.
+	// See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html for general
+	// information about configuration, characteristics, and limitations of Elastic IP addresses.
+	// +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+	// +optional
+	// +listType=atomic
+	// Each entry in the eipAllocations list must be unique
+	// +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x == y))`,message="eipAllocations cannot contain duplicates"
+	// A maximum of 10 IP allocations are permitted
+	// +kubebuilder:validation:MaxItems=10
+	EIPAllocations []EIPAllocation `json:"eipAllocations"`
 }
 
+// EIPAllocation is a reference to an EIP that can be allocated to an ELB in the AWS environment.
+// The pattern for an EIP allocation ID is as follows:
+//
+// Starts with eipalloc-
+//
+// # Followed by an alphanumeric string of exactly 17 characters
+//
+// + Explanation of the regex `^eipalloc-[0-9a-fA-F]{17}$` for validating value of the EIPAllocation:
+//
+// + ^eipalloc- ensures the string starts with "eipalloc-".
+//
+// + [0-9a-fA-F]{17} matches exactly 17 hexadecimal characters (0-9, a-f, A-F).
+//
+// + $ ensures the string ends after the 17 hexadecimal characters.
+//
+// + Example of Valid and Invalid values:
+//
+// + eipalloc-1234567890abcdef1 is valid.
+//
+// + eipalloc-1234567890abcde is not valid (too short).
+//
+// + eipalloc-1234567890abcdefg is not valid (contains a non-hex character 'g').
+//
+// + Max length is calculated as follows:
+//
+// + eipalloc- = 9 chars and 17 hexadecimal chars after `-`
+//
+// + So, total is 17 + 9 = 26 chars required for value of an EIPAllocation.
```

Comment:
Sucking eggs here as well, I don't think we need this much detail

Reply:
do you want to remove the whole part ?

Reply:
just keeping it as it is as discussed above as it is a developer doc.